master/internal/user/service.go

```diff
@@ -203,17 +204,28 @@ func (s *Service) ProcessProxyAuthentication(c echo.Context) (done bool, err err
 		return true, redirectToLogin(c)
 	}
 
-	switch user, _, err := UserByToken(token, s.extConfig); err {
-	case nil:
-		if !user.Active {
-			return true, redirectToLogin(c)
-		}
-		return false, nil
-	case db.ErrNotFound:
+	user, _, err := UserByToken(token, s.extConfig)
+	if errors.Is(err, db.ErrNotFound) {
 		return true, redirectToLogin(c)
-	default:
+	} else if err != nil {
 		return true, err
 	}
+	if !user.Active {
+		return true, redirectToLogin(c)
+	}
+
+	taskID := c.Param("service")
+	ownerID, err := db.GetCommandOwnerID(c.Request().Context(), model.TaskID(taskID))
```

Comment:
Is it correct to say this proxy only auth only used by notebooks and tensorboards now? 

I tried to dig through the code but couldn't really tell if I was missing other use cases of this proxy auth.

Reply:
hmm yes, also shells. 

when we'll be shipping "heroes of determined 3" / generic proxies, this proxy will be used by any task type, but that's future work.